bdk_rpc_wallet_example/src/rpc.rs

```diff
@@ -0,0 +1,314 @@
+            "main" => Network::Bitcoin,
+            "test" => Network::Testnet,
+            "regtest" => Network::Regtest,
+            "signet" => Network::Signet,
+            _ => return Err(RpcError::General("Invalid Network string".to_string())),
+        };
+        if network != config.network {
+            return Err(RpcError::General("Invalid Network".to_string()));
+        }
+
+        import_multi(
+            &client,
+            txout_index
+                .inner()
+                .all_spks()
+                .iter()
+                .map(|(_, script)| script),
+        )?;
+
+        Ok(Self { client })
+    }
+
+    fn get_tip(&self) -> Result<(u64, BlockHash), RpcError> {
+        Ok(self
+            .client
+            .get_blockchain_info()
+            .map(|i| (i.blocks, i.best_block_hash))?)
+    }
+
+    /// Scan for a keychain tracker, and create an initial [`SparseChain`] candidate update.
+    /// This update will only contain [`Txid`]s in SparseChain, and no actual transaction data.
+    ///
+    /// User needs to fetch the required transaction data and update the [`SparseChain`] before applying it.
+    pub fn wallet_scan(
+        &self,
+        local_chain: &BTreeMap<u32, BlockHash>,
+    ) -> Result<SparseChain, RpcError> {
+        let mut sparse_chain = SparseChain::default();
+
+        let mut last_common_height = 0;
+        for (&height, &original_hash) in local_chain.iter().rev() {
+            let update_block_id = BlockId {
+                height,
+                hash: self.client.get_block_hash(height as u64)?,
+            };
+            let _ = sparse_chain
+                .insert_checkpoint(update_block_id)
+                .expect("should not collide");
+            if update_block_id.hash == original_hash {
+                last_common_height = update_block_id.height;
+                break;
+            }
+        }
+
+        // Insert the new tip so new transactions will be accepted into the sparse chain.
+        let tip = self.get_tip().map(|(height, hash)| BlockId {
+            height: height as u32,
+            hash,
+        })?;
+        if let Err(failure) = sparse_chain.insert_checkpoint(tip) {
+            match failure {
+                sparse_chain::InsertCheckpointError::HashNotMatching { .. } => {
+                    // There has been a re-org before we even begin scanning addresses.
+                    // Just recursively call (this should never happen).
+                    return self.wallet_scan(local_chain);
+                }
+            }
+        }
+
+        // Fetch the transactions
+        let page_size = 1000; // Core has 1000 page size limit
+        let mut page = 0;
+
+        let _ = self.rescan_blockchain(Some(last_common_height as usize), None);
+
+        let mut txids_to_update = Vec::new();
+
+        loop {
+            let list_tx_result = self
+                .list_transactions(None, Some(page_size), Some(page * page_size), Some(true))?
```

Comment:
wasn't the plan to use the label arugment (the first one) to differentiate between wallets (I was getting confused between this and wallet name).